src/bitcoin-chainstate.cpp

```diff
@@ -49,7 +51,7 @@ int main(int argc, char* argv[])
     SelectParams(CBaseChainParams::MAIN);
     const CChainParams& chainparams = Params();
 
-    init::SetGlobals(); // ECC_Start, etc.
+    kernel::Context kernel_context{};
```

Comment:
This also needs a `kernel::SanityChecks(kernel_context)`, no? Another argument for making it part of context creation.

(It was also missing before, but presumably because it was still tangled up with the ui).

Reply:
Btw, I'm loving how this just untangled itself. Great suggestion from @ryanofsky [here](https://github.com/bitcoin/bitcoin/pull/25065#discussion_r882101216)!

Reply:
Fixed as of: fa2ee7b12c0c15be8e9807444443b5bd2c91c3e5

Reply:
Completely agree it's wonky, I had a similar complaint here: https://github.com/bitcoin/bitcoin/pull/25065#discussion_r883689055

I think it's fine as-is for now though, as context creation will definitely be changing in subsequent PRs.